### PR TITLE
Add timeout middleware.

### PR DESCRIPTION
**Pull request by myronmarston**

### CODE REVIEW

File: spec/integration/middleware/timeout_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+require 'byebug'
```
Comment (vadim-moz) [inline spec/integration/middleware/timeout_spec.rb]: do we need `byebug`?

File: lib/qless/middleware/timeout.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    # and the block should return the desired timeout in seconds.
+    # This allows you to set a hard constant time out to a particular job class
+    # (using something like `extend Qless::Middleware::Timeout.new { 60 * 60 }`),
+    # or a variable timeout based on the individual TTLs of each job
+    # (using something like `extend Qless::Middleware::Timeout.new { |job| job.ttl * 1.1 }`).
+    class Timeout < Module
+      def initialize
+        module_eval do
+          define_method :around_perform do |job|
+            timeout_value = yield job
```
Comment (vadim-moz) [inline lib/qless/middleware/timeout.rb]: I am not sure if this syntax is compatible with older versions of Ruby (Qless still supports 1.9.3). Please do not bother to comment if it does.

**Comment by vadim-moz** [comment]
LGTM (nitpicks aside)

File: lib/qless/middleware/timeout.rb (hunk 2 above)
Comment (myronmarston) [inline lib/qless/middleware/timeout.rb]: The travis build will tell us if it doesn't :).

BTW, which part of the syntax are you unsure about?

File: spec/integration/middleware/timeout_spec.rb (hunk 1 above)
Comment (myronmarston) [inline spec/integration/middleware/timeout_spec.rb]: Nope, I didn't mean to leave it.

**Pull request merged**